code/category/test_script_for_concept_catgorization.py

```diff
@@ -0,0 +1,164 @@
+# from code.models import models
+import random
+from sklearn.cluster import KMeans
+from collections import Counter
+
+
+class concept_categorization:
+    def __init__(self):
+        self.member = []
+
+    def add_member(self, member_list):
+        self.member.append(member_list)
+
+    def get_score(self):
+        correct = 0
+        for i in range(len(self.member)):
+            if self.member[i][0]==self.member[i][2]:
+                correct+=1
+        return correct/len(self.member)
+
+class embedding_model:
+
+    def __init__(self, inputModel):
+        print("load model ...")
+        #self.model = models.W2V.fasttext_from_text(inputModel)
+
+    def get_word2vec(word , inputModel):
+        # vec = list(self.model.getVec(word))
+        # return vec
+        return [random.randint(1,100),random.randint(1,100),random.randint(1,100)]
+
+def k_means(input_vectors , K):
+    kmeans = KMeans(n_clusters=K)
+    kmeansoutput = kmeans.fit_predict(input_vectors)
+    output_list = list(kmeansoutput)
+    return output_list
+
+def write_in_file(path,data) :
+    with open(path, 'w' , encoding='utf-8') as f:
+        f.write(data)
+
+def preprocessing(input_file,input_model):
+
+    word_embeddings = {}
+    concept_categorization={}
+    members = []
+    concept_counter=-1
+    concept_number={}
+
+    data = open(input_file, "r", encoding='utf-8')
+    lines = data.readlines()
+    unique_concepts = []
+    result=[]
+
+    for item in lines:
+
+        item=item.split("\n")[0]
+        cnp = item.split(",")[0]
+        mbr = item.split(",")[1]
+        members.append(mbr.strip())
+
+        if cnp not in unique_concepts:
+            current_members=[]
+            unique_concepts.append(cnp)
+            concept_number.update([(cnp, str(concept_counter))])
+
+            current_members.append(mbr)
+            concept_categorization.update([(str(cnp), current_members)])
+
+            concept_counter+=1
+        else:
+            current_members.append(mbr)
+            concept_categorization.update([(str(cnp), current_members)])
+
+        mach = [mbr , concept_counter]
+
+        result.append(mach)
+
+    vectors = []
+    word2vec = embedding_model(input_model)
```

Comment:
Same as above if you accept W2V class as parameter of this method then everything will be simple